src/crimson/osd/heartbeat.cc

```diff
@@ -184,9 +184,19 @@ void Heartbeat::update_peers(int whoami)
   }
 }
 
+Heartbeat::osds_t Heartbeat::get_peers() const
+{
+  osds_t osds;
+  osds.reserve(peers.size());
+  for (auto& peer : peers) {
+    osds.push_back(peer.first);
+  }
+  return osds;
+}
+
 void Heartbeat::remove_peer(osd_id_t peer)
 {
-  auto found = peers.find(peer);
+  [[maybe_unused]] auto found = peers.find(peer);
```

Comment:
`peers` is an instance of `std::map<osd_id_t, Peer>`, so maybe let's exchange the current assert with:
```cpp
  assert(peers.count(peer) == 1);
```

Reply:
@rzarzynski updated and repushed.